controllers/volsync/vshandler.go

```diff
@@ -625,7 +627,7 @@ func (v *VSHandler) validatePVCAndAddVRGOwnerRef(pvcName string) (*corev1.Persis
 func (v *VSHandler) validateSecretAndAddVRGOwnerRef(secretName string) (bool, error) {
 	secret := &corev1.Secret{}
 
-	err := v.client.Get(v.ctx,
+	err := v.APIReader.Get(v.ctx,
```

Comment:
The reconciler code should use only the client passed. This is to avoid load on the API server by making multiple requests to the same.

IOW, in the normal controller logic the `client` passed in would be the caching client, and any errors would be retried in the next reconcile loop as usual, and at some point this would find the resource it is looking for in the local cache.

From the test code, we can pass this `client` as a non-caching client, such that it is not waiting on the cache to be up to date for the operation to succeed.

Current change impacts the controller reconciles by reaching out to the API server more frequently, which we should avoid as a practice.